### ISSUE TITLE
IO errors are not informative without the path that caused them

### ISSUE THREAD

**Issue by tcharding** [comit-network/comit-rs#1582]
## Problem

When we get errors caused by IO on the seed file the IO error is printed to stderr.  Rust IO errors do not contain the path that caused the error so this output is totally un-informative.

```
cnd --seed-file /some/non/existant/path
...
Error: seed: io error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

## goal

Have the path that caused an error printed to stderr along with the error (reason).

```
cnd --seed-file /some/non/existant/path
...
Error: seed: io error: Os { code: 2, kind: NotFound, message: "No such file or directory: /some/non/existant/path" }
```

## Tips

This is widely discussed in the Rust community, what ever solution we come up with we should probably do company wide i.e., this will probably need some consideration with team members who care about such things.

This is related to, or could possibly be done at a similar time to, https://github.com/comit-network/comit-rs/issues/487

**Comment by da-kami** [comment]
@tcharding please provide examples in the ticket (how does the error look now, how would it look in the future)

Recommendation:
Investigate this together with logging spike #487

**Comment by D4nte** [comment]
> @tcharding please provide examples in the ticket (how does the error look now, how would it look in the future)
> 
> Recommendation:
> Investigate this together with logging spike #487

`Cannot read seed file` <- does not print seed path.

**Comment by thomaseizinger** [comment]
I'd recommend we go for a combination of the `err-derive` and `anyhow` crates. To me, that sounds like the most appealing solution to error handling :)

For this particular issue, `anyhow` is sufficient though. Introducing `err-derive` is useful in general.

**Comment by tcharding** [comment]
The anyhow crate resolves this issue. We have started using it already (see cnd/src/db/mod.rs).  

From the docs (linked below)
```
use anyhow::{Context, Result};

fn main() -> Result<()> {
    ...
    it.detach().context("failed to detach the important thing")?;

    let content = std::fs::read(path)
        .with_context(|| format!("failed to read instrs from {}", path))?;
    ...
}
```


ref: https://docs.rs/anyhow/1.0.19/anyhow/
